pkg/cimfs/mount_cim.go

```diff
@@ -118,3 +118,32 @@ func MountMergedBlockCIMs(mergedCIM *BlockCIM, sourceCIMs []*BlockCIM, mountFlag
 	}
 	return fmt.Sprintf("\\\\?\\Volume{%s}\\", volumeGUID.String()), nil
 }
+
+// Mounts a verified block CIM with the provided root hash. The root hash is usually
+// returned when the CIM is sealed or the root hash can be queries from a block CIM.
+// Every read on the mounted volume will be verified to match against the provided root
+// hash if it doesn't, the read will fail.  The CIM MUST have been created with the
+// verified creation flag.
+func MountVerifiedBlockCIM(bCIM *BlockCIM, mountFlags uint32, volumeGUID guid.GUID, rootHash []byte) (string, error) {
```

Comment:
Example from Go Standard Library
The crypto/sha256 package uses [32]byte for a SHA-256 hash to enforce size at compile time.

In Your Case
Since your function already checks length and the hash must be exactly cimHashSize, using [cimHashSize]byte would provide a stronger guarantee and may be preferable if all your call sites can easily use arrays. Otherwise, stick with []byte for flexibility.

Reply:
In case of CimFS, the root digest is going to be 32 bytes for now (first 32 bytes of the SHA512 of the block CIM header). But in future it may change to using the 40 or 64 bytes of that hash, so if we have a slice the API won't break then. Let me know what you think.